client/src/modules/reports/generate/stock_entry/stock_entry.config.js

```diff
@@ -0,0 +1,97 @@
+angular.module('bhima.controllers')
+  .controller('stock_entryController', StockEntryConfigController);
+
+StockEntryConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state',
+  'LanguageService',
+];
+
+function StockEntryConfigController($sce, Notify, SavedReports, AppCache, reportData, $state, Languages) {
+  const vm = this;
+  const cache = new AppCache('configure_stock_entry_report');
+  const reportUrl = 'reports/stock/entry';
+
+  // default values
+  vm.includePurchaseEntry = 1;
```

Comment:
Just FYI - most reports put their default values in an object, usually called `reportDetails`.  This allows us to really easily merge default values with current values using `Object.assign()`.  Here is how that works:
```js
var userOptions = { date : Date('2015-03-01'), includeDonation : 0 };
var defaults = { includePurchaseEntry : 1, includeDonation : 1 };

var selected = Object.assign({}, defaults, userOptions);

// selected is now:
// { date : Date('2015-03-01'), includeDonation : 0, includePurchaseEntry : 1 }
```
You can try it in your console!

My proposal is to wrap all these `vm.include*` variables in an object called `vm.reportDetails`.  Like this:
```js
vm.reportDetails = {
  includeIntegrationEntry : 0,
  // etc...
};
```

Reply:
Did you want to change this?  I'm fine either way.